src/e2e-browser/e2e.testcafe.ts

```diff
@@ -45,7 +47,31 @@ config({
   silent: process.env.CI === "true",
 });
 
-fixture("End-to-end tests").page("http://localhost:1234/end-to-end-test.html");
+const requestLogger = RequestLogger(undefined, {
+  logRequestHeaders: true,
+  logRequestBody: true,
+  logResponseHeaders: true,
+  logResponseBody: true,
+  stringifyRequestBody: true,
+  stringifyResponseBody: true,
+});
+
+fixture("End-to-end tests")
+  .page("http://localhost:1234/end-to-end-test.html")
+  .requestHooks(requestLogger)
+  .after(async () => {
+    await promises.mkdir(resolve(__dirname, "testcafe-requests"), {
+      recursive: true,
+    });
+    await promises.writeFile(
+      resolve(
+        __dirname,
+        "testcafe-requests",
+        `testcafe-requests-${Date.now()}.log`
+      ),
+      JSON.stringify(requestLogger.requests, undefined, 2)
```

Comment:
Yeah, this capability (capturing requests/responses) is super-valuable I think. But when I do this on AuthN though it's spewing out 350MB per run! It's easy to tweak the output of course though - which I did for AuthN, and some very interesting insights in the continuing intermittent failures, which we can discuss offline (i.e., I'd suggest reducing the output being generated here before merging).

Reply:
The TestCafe tests for solid-client are pretty limited, so it's about 3MB per browser/OS combo (so with three Windows browsers, two on Linux and one on MacOS, that comes down to 18MB). That sounds pretty manageable, so I'd propose not throwing away data here.